### PR TITLE
FUSETOOLS-2237 - Support runtime providers on custom Palette entries

### PR DESCRIPTION
**Pull request by apupier**
- API for the Custom PaletteEntry updated
- updated custom implementations: ActiveMQ and camel-dozer

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateEndpointFigureFeature.java [hunk 1]
```diff
@@ -94,4 +94,13 @@ protected String getIconName() {
 		}
 		return super.create(context);
 	}
+
+	/**
+	 *  /!\ Public for test purpose only
+	 * @return
+	 */
+	@Deprecated
+	public List<Dependency> getDeps() {
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateEndpointFigureFeature.java]: I know its marked deprecated but can we still use better naming for that method like getDependencies?

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java [hunk 2]
```diff
@@ -18,61 +18,81 @@
...
 	private static final String PROTOCOL = "activemq";
 	
-    /* (non-Javadoc)
-     * @see org.fusesource.ide.camel.editor.provider.ICustomPaletteEntry#newCreateFeature(org.eclipse.graphiti.features.IFeatureProvider)
-     */
     @Override
     public ICreateFeature newCreateFeature(IFeatureProvider fp) {
-        return new CreateEndpointFigureFeature(fp, "ActiveMQ", "Creates an ActiveMQ endpoint...", "activemq:queue:foo", getRequiredDependencies());
+        List<Dependency> requiredDependencies = getRequiredDependencies(CamelUtils.getRuntimeProvider(fp));
+		return new CreateEndpointFigureFeature(fp, ACTIVE_MQ, "Creates an ActiveMQ endpoint...", "activemq:queue:foo", requiredDependencies);
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java]: indentation?

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java [hunk 3]
```diff
@@ -18,61 +18,81 @@
...
-        		"org.apache.camel",
-        		"camel-jms",
-        		CamelUtils.getCurrentProjectCamelVersion()));
+        deps.add(createActiveMQDependency(runtimeProvider));
+        deps.add(createJMSDependency(runtimeProvider));
         return deps;
     }
 
-    private Dependency createDependency(String groupId, String artifactId, String version) {
+	private Dependency createJMSDependency(String runtimeProvider) {
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java]: indentation? looks like the whole class needs a reformat.

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java [hunk 4]
```diff
@@ -18,61 +18,81 @@
...
+
+	private Dependency createActiveMQDependency(String runtimeProvider) {
+		ActiveMQPaletteEntryDependenciesManager activeMQPaletteEntryDependenciesManager = new ActiveMQPaletteEntryDependenciesManager();
+		return createDependency(
+        		ActiveMQPaletteEntryDependenciesManager.ORG_APACHE_ACTIVEMQ,
+        		activeMQPaletteEntryDependenciesManager.getArtifactId(runtimeProvider),
+        		activeMQPaletteEntryDependenciesManager.getActiveMQVersion(getCurrentProjectCamelVersion()));
+	}
+
+	String getCurrentProjectCamelVersion() {
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java]: can we make it private or is it used in the same package again?

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java [hunk 5]
```diff
@@ -18,61 +18,81 @@
...
+		return CamelUtils.getCurrentProjectCamelVersion();
+	}
+
+	private Dependency createDependency(String groupId, String artifactId, String version) {
     	Dependency dep = new Dependency();
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java]: indentation

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntryDependenciesManager.java [hunk 6]
```diff
@@ -17,28 +17,30 @@
...
 	public static final String RED_HAT_SUFFIX = ".redhat-";
-	public static final Map<String, String> camelToAMQVersionMapping;
+	static final String ORG_APACHE_ACTIVEMQ = "org.apache.activemq";
+	public static final String ACTIVEMQ_CAMEL = "activemq-camel";
+	public static final String ACTIVEMQ_CAMEL_STARTER = "activemq-camel-starter";
+	
 	// TODO: change me after each release
 	public static final String LATEST_AMQ_VERSION = "5.11.0";
-	
+	public static final Map<String, String> camelToAMQVersionMapping;
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntryDependenciesManager.java]: can we use upper case for constants?

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/extensions/DataTransformationPaletteEntry.java [hunk 7]
```diff
@@ -17,53 +17,56 @@
...
-     * #newCreateFeature(org.eclipse.graphiti.features.IFeatureProvider)
-     */
     @Override
     public ICreateFeature newCreateFeature(IFeatureProvider fp) {
-        return new DataMapperEndpointFigureFeature(fp,
+        List<Dependency> requiredDependencies = getRequiredDependencies(CamelUtils.getRuntimeProvider(fp));
+		return new DataMapperEndpointFigureFeature(fp,
                 Messages.DataTransformationPaletteEntry_paletteName,
-                Messages.DataTransformationPaletteEntry_paletteDescription, getRequiredDependencies());
+                Messages.DataTransformationPaletteEntry_paletteDescription, requiredDependencies);
```
Comment (lhein) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/extensions/DataTransformationPaletteEntry.java]: indentation?

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/extensions/DataTransformationPaletteEntry.java [hunk 8]
```diff
@@ -17,53 +17,56 @@
...
-        dep.setGroupId("org.apache.camel"); //$NON-NLS-1$
-        dep.setArtifactId("camel-dozer"); //$NON-NLS-1$
+        dep.setGroupId(ORG_APACHE_CAMEL);
+        dep.setArtifactId(computeArtifactId(runtimeProvider));
         dep.setVersion(CamelUtils.getCurrentProjectCamelVersion());
         deps.add(dep);
         return deps;
     }
+
+	private String computeArtifactId(String runtimeProvider) {
```
Comment (lhein) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/extensions/DataTransformationPaletteEntry.java]: indentation

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java (hunk 4 above)
Comment (apupier) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ActiveMQPaletteEntry.java]: it is used in Unit tests

**Pull request merged**